pkg/controller/propagator/propagation.go

```diff
@@ -43,88 +273,110 @@ func (r *ReconcilePolicy) handleRootPolicy(instance *policiesv1.Policy) error {
+		)
+
 		if err != nil {
-			// there was an error, requeue
-			reqLogger.Error(err, "Failed to list replicated policy...")
+			reqLogger.Info("Giving up on the policy clean up...")
+			r.recordWarning(instance, "One or more replicated policies could not be deleted")
 			return err
 		}
-		for _, plc := range replicatedPlcList.Items {
-			// #nosec G601 -- no memory addresses are stored in collections
-			err := r.client.Delete(context.TODO(), &plc)
-			if err != nil && !errors.IsNotFound(err) {
-				reqLogger.Error(err, "Failed to delete replicated policy...", "Namespace", plc.GetNamespace(),
-					"Name", plc.GetName())
-				return err
-			}
-		}
+
 		r.recorder.Event(instance, "Normal", "PolicyPropagation",
 			fmt.Sprintf("Policy %s/%s was disabled", instance.GetNamespace(), instance.GetName()))
 	}
-	// get binding
+
+	// Get the placement binding in order to later get the placement decisions
 	pbList := &policiesv1.PlacementBindingList{}
-	err := r.client.List(context.TODO(), pbList, &client.ListOptions{Namespace: instance.GetNamespace()})
+	err := retry.Do(
+		func() error {
+			return r.client.List(
+				context.TODO(), pbList, &client.ListOptions{Namespace: instance.GetNamespace()},
+			)
+		},
+		getRetryOptions(reqLogger, "Retrying to list the placement bindings...")...,
+	)
+
 	if err != nil {
-		reqLogger.Error(err, "Failed to list pb...")
+		reqLogger.Info("Giving up on listing the placement bindings...")
+		r.recordWarning(instance, "Could not list the placement bindings")
 		return err
 	}
-	// get placement
-	placement := []*policiesv1.Placement{}
-	// a set in the format of `namespace/name`
-	allDecisions := map[string]struct{}{}
-	for _, pb := range pbList.Items {
-		subjects := pb.Subjects
-		for _, subject := range subjects {
-			if subject.APIGroup == policiesv1.SchemeGroupVersion.Group &&
-				subject.Kind == policiesv1.Kind && subject.Name == instance.GetName() {
-				decisions, p, err := getPlacementDecisions(r.client, pb, instance)
-				if err != nil {
-					return err
-				}
-				placement = append(placement, p)
-				// only handle replicate policy when policy is not disabled
-				if !instance.Spec.Disabled {
-					// plr found, checking decision
-					for _, decision := range decisions {
-						key := fmt.Sprintf("%s/%s", decision.ClusterNamespace, decision.ClusterName)
-						allDecisions[key] = struct{}{}
-						// create/update replicated policy for each decision
-						err := r.handleDecision(instance, decision)
-						if err != nil {
-							return err
-						}
-					}
-				}
-				// only handle first match in pb.spec.subjects
-				break
-			}
-		}
+
+	// allDecisions and failedClusters are sets in the format of <namespace>/<name>
+	placements, allDecisions, failedClusters, allFailed := r.handleDecisions(instance, pbList)
+	if allFailed {
+		reqLogger.Info("Failed to get any placement decisions. Giving up...")
+		r.recordWarning(instance, "Could not get the placement decisions")
+		return err
```

Comment:
@mprahl :  where  is this err   set ?  the previously set value for err is already being returned. may be i am not reading it correctly but i dont see where its set again.

Reply:
Thanks for catching that! It was a copy/paste error. I'll fix it in the next push.